pkg/listwatch/listwatch.go

```diff
@@ -199,9 +202,9 @@ func newMultiWatch(lws []cache.ListerWatcher, resourceVersions []string, options
 
 	wg.Add(len(lws))
 
-	for i, lw := range lws {
+	for _, lw := range lws {
 		o := options.DeepCopy()
-		o.ResourceVersion = resourceVersions[i]
+		o.ResourceVersion = resourceVersions[0]
```

Comment:
This will produce odd results if more than one listerwatcher is given since they'll now be watching from the wrong version. Why not enforce that lws has length == 1? Or that it is a single instance vs a slice?

Reply:
> Or that it is a single instance vs a slice?

Yes exactly we could simply not pass in a slice and instead a single instance and refactor the entire function, but as mentioned in the description I did not want to focus too much on refactoring the entire thing as this fixes a bug in 0.40 version, but in master as per the linked issue we want to move away from custom multi listwatcher as we had a few bugs with it and it has known issues as we know now as from the research Paul did #3218. Hope that clarifies it? :)

Reply:
I'm perfectly happy with a short term fix, just the string slice should be a single string, and we have a single call site, so it's a 3 line change from this one :)

Reply:
Agreed, as mentioned, let's make this a single string vs a slice to avoid confusion and unexpected results/usages

Reply:
Sure will make the change, but FYI it's not a 3 line change from this one ;)

Reply:
My apologies, if it does turn out to be way to complicated I'm happy with not doing it, just wanted us to try.

Reply:
No worries, just need to adjust all the unit tests and retest on vanilla k8s and OpenShift clusters and CI so might take a day or two longer depending on CI. :) WIP until then.

Reply:
just chiming in that my fear is that without this last change we are leaving the library in a particularly weird place where it doesn't attempt to do the things it is supposedly designed for (even if the original impl has bugs). I am worried that this will be confusing and could lead to future issues or misunderstandings

Reply:
understandable, I'm not too worried since this is just on the release branch, and not into master, but leaving it up to Lili to judge the trade-offs, I'm happy with going either way